lib/sqlalchemy/dialects/postgresql/pg8000.py

```diff
@@ -153,6 +153,40 @@ def process(value):
             return process
 
 
+class _PGVarchar(sqltypes.VARCHAR):
+    def bind_processor(self, dialect):
+        proc = super(_PGVarchar, self).bind_processor(dialect)
+        if proc is None:
+            return proc
+        else:
+            pg_varchar = dialect.dbapi.PGVarchar
+
+            def new_proc(value):
+                res = proc(value)
+                if res is None:
+                    return res
+                else:
+                    return pg_varchar(value)
+            return new_proc
+
+
+class _PGText(sqltypes.TEXT):
+    def bind_processor(self, dialect):
+        proc = super(_PGText, self).bind_processor(dialect)
+        if proc is None:
+            return proc
+        else:
+            pg_text = dialect.dbapi.PGText
+
+            def new_proc(value):
+                res = proc(value)
+                if res is None:
+                    return res
+                else:
+                    return pg_text(value)
```

Comment:
I would assume this is a wrapper that allows pg8000 to know more about the type for strings coming in?   is this needed for python 3 or is this a compatibility thing for python 2?    also is there any difference between text/varchar, my understanding is that postgresql itself sees no difference between these two datatypes.